### PR TITLE
Feature/hexdump

### PR DESCRIPTION
**Pull request by peter50216**

### CODE REVIEW

File: lib/pwnlib/pwn.rb [hunk 1]
```diff
@@ -7,16 +7,19 @@
...
 
 # include this module in a class to use all pwnlib functions in that class
 # instance.
 module Pwn
   include ::Pwnlib::Context
 
-  include ::Pwnlib::Util::Packing::ClassMethod
-  include ::Pwnlib::Util::Cyclic::ClassMethod
-  include ::Pwnlib::Util::Fiddling::ClassMethod
+  include Pwnlib::Util::Cyclic::ClassMethod
```
Comment (david942j) [inline lib/pwnlib/pwn.rb]: OAO?

File: lib/pwnlib/util/hexdump.rb [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+        # @param [#read] io
+        #   The object to be dumped.
+        # @!macro hexdump_options
+        # @return [Enumerator<String>]
+        #   The resulting hexdump, line by line.
+        def hexdump_iter(io, width: 16, skip: true, offset: 0, style: {}, highlight: '')
+          Enumerator.new do |y|
+            style = DEFAULT_STYLE.merge(style)
+            highlight.bytes.each { |b| style[b] = HIGHLIGHT_STYLE }
+            (0..255).each do |b|
```
Comment (david942j) [inline lib/pwnlib/util/hexdump.rb]: `256.times` (?

File: test/util/hexdump_test.rb [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+    )
+  end
+
+  def test_highlight
+    orig_verbose = $VERBOSE
+    orig_style = HIGHLIGHT_STYLE
+    begin
+      $VERBOSE = nil
+      ::Pwnlib::Util::HexDump::ClassMethod.const_set(:HIGHLIGHT_STYLE, ->(s) { "#{s}!" })
+      assert_lines_equal(<<-'EOS', hexdump('abcdefghi', highlight: 'orange'))
```
Comment (david942j) [inline test/util/hexdump_test.rb]: `highlight: 'orange'`  is confusing
maybe use `highlight: 'aehi'` ?

File: lib/pwnlib/util/hexdump.rb (hunk 2 above)
Comment (david942j) [inline lib/pwnlib/util/hexdump.rb]: ok

File: test/util/hexdump_test.rb (hunk 3 above)
Comment (peter50216) [inline test/util/hexdump_test.rb]: done. Just want to write orange somewhere :P

File: lib/pwnlib/util/hexdump.rb [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    # @todo Control coloring by context?
+    # @example Call by specifying full module path.
+    #   require 'pwnlib/util/fiddling'
+    #   Pwnlib::Util::Fiddling.enhex('217') #=> '323137'
+    # @example require 'pwn' and have all methods.
+    #   require 'pwn'
+    #   enhex('217') #=> '323137'
+    module HexDump
+      # @note Do not create and call instance method here. Instead, call module method on {HexDump}.
+      module ClassMethod
```
Comment (david942j) [inline lib/pwnlib/util/hexdump.rb]: `ClassMethods` :p

**Comment by david942j** [comment]
LGTM

**Pull request merged**